### PR TITLE
[Connector API] Allow Connector status transition to the same status for CONNECTED, CONFIGURED, ERROR

### PR DESCRIPTION
**Pull request by jedrazb**
## Changes

Update valid status transitions for `Connector`. We should allow transition to same status for following statuses:
- `CONNECTED` (e.g. starting a [sync](https://github.com/elastic/connectors/blob/main/connectors/protocol/connectors.py#L742) if connector status is already `connected`)
- `CONFIGURED` (when sending multiple reqs to update configuration)
- `ERROR` (when multiple concurrent running jobs report error, e.g. content sync and access control job)

## Validations

- unit tests
- manual test

### CODE REVIEW

**Comment by jedrazb** [comment]
@elasticmachine merge upstream

**Review by artem-shelkovnikov — COMMENTED** [review]
While it makes sense, I'm hesitant - some transitions are natural (CONNECTED -> CONNECTED is okay), but some are... weird?

CREATED -> CREATED - can it even happen?
ERROR -> ERROR - I guess it's when connector consistently errors out?
CONFIGURED -> CONFIGURED - I guess it happens if you edit configuration and then quickly edit it again, before it's transitioned to CONNECTED?

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorStateMachine.java [hunk 1]
```diff
@@ -56,6 +58,9 @@ public static void assertValidStateTransition(ConnectorStatus current, Connector
     }
 
     public static Set<ConnectorStatus> validNextStates(ConnectorStatus current) {
-        return VALID_TRANSITIONS.getOrDefault(current, Collections.emptySet());
+        Set<ConnectorStatus> nextStates = new HashSet<>(VALID_TRANSITIONS.getOrDefault(current, Collections.emptySet()));
+        // Transition to the same state is allowed
+        nextStates.add(current);
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorStateMachine.java:63]: Why not just add the explicit states in VALID_TRANSITIONS?
This obfuscates what are actually valid transitions by splitting the logic between that literal map and this function.

**Comment by jedrazb** [comment]
Thank you both for reviewing, yeah it makes sense to spend a bit of time to explicitly state this in the transition table. Let me update the PR :)

**Comment by jedrazb** [comment]
@elasticmachine merge upstream

**Comment by jedrazb** [comment]
@elasticmachine merge upstream

**Comment by jedrazb** [comment]
@seanstory @artem-shelkovnikov I updated the PR. The changes are now to allow transition to same status for following statuses:

- CONNECTED (e.g. starting a [sync](https://github.com/elastic/connectors/blob/main/connectors/protocol/connectors.py#L742) if connector status is already connected)
- CONFIGURED (when sending multiple reqs to update configuration)
- ERROR (when multiple concurrent running jobs report error, e.g. content sync and access control job)

I'm changing this just for connectors. I don't think SyncJob logic needs any changes

**Review by seanstory — APPROVED** [review]
LGTM

**Pull request merged**